writer.go

```diff
@@ -1581,11 +1581,19 @@ func (c *writerColumn) recordPageStats(headerSize int32, header *format.PageHead
 
 			if existingMaxValue.isNull() || c.columnType.Compare(maxValue, existingMaxValue) > 0 {
 				buf := c.columnChunk.MetaData.Statistics.MaxValue[:0]
+				// if maxValue is empty string, c.columnChunk.MetaData.Statistics.MaxValue should be []bytes{}, but nil
+				if buf == nil && maxValue.Kind() == ByteArray && len(maxValue.ByteArray()) == 0 {
```

Comment:
`len(maxValue.ByteArray())` seems like it would be a good opportunity to introduce a `Len() int` method on `parquet.Value`... work for another day!